### ISSUE TITLE
Should the image input to the neural network be before or after dedistortion?

### ISSUE THREAD

**Issue by vtasStu** [apple/ml-depth-pro#33]
Thank you for your great work. I have a few questions.
1. Should the image be distorted?
2. How do you consider the intrinsic cx and cy? Will the offset of cx and cy from the center of the image affect the depth?

**Comment by xiaodongww** [comment]
Hi，I also have the same quesiton. By the way, intrinsics usuall have four parameters (fx, fx, cx, cy), is only f_px enough to produce metric depth? do you have any idea? @vtasStu

**Comment by JVPC0D3R** [comment]
@xiaodongww I used both the fx parameter and a mean between fx and fy. None of them gave me a proper depth map for my custom images. 

```
# horizontal focal length
f_px= torch.tensor(fx)
image = transform(frame)
```

```
# mean focal length
f_px= torch.tensor((fx + fy) /2)
image = transform(frame)
```

After processing the image and getting the metric depth map I ended up with the map below. I tested the same code with several samples and I only got shapes of the furthest elements represented in the map.

![image](https://github.com/user-attachments/assets/52f368e1-21c5-4327-8bd0-dec4c410cba4)

**Comment by JVPC0D3R** [comment]
The depth map is correct, the outputs go really far away in this example (up to 10km). In order to visualize the output in a shorter range you can use ```depth = depth.clip(0,200)```

Regarding the original question, I recomend undistort your image if possible.

**Comment by GreaBugs** [comment]
> The depth map is correct, the outputs go really far away in this example (up to 10km). In order to visualize the output in a shorter range you can use `depth = depth.clip(0,200)`
> 
> Regarding the original question, I recomend undistort your image if possible.

Depthpro should have done interpolation, which means using the resized image to predict the FOV of the original image? How should we understand this?
